query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+            + "and ( start_time_millis > 1570658506605 and end_time_millis < 1570744906673 )"
+            + " group by span_id order by count(distinct span_id) limit 15",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithStringArray() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createColumnExpression("Span.id").build());
+
+    String spanId1 = "042e5523ff6b2506";
+    String spanId2 = "041e5523ff6b2501";
+    Filter filter = createInFilter("Span.id", List.of(spanId1, spanId2));
+    builder.setFilter(filter);
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "SELECT encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND span_id IN ("
+            + "decode('"
+            + spanId1
+            + "', 'hex'), "
+            + "decode('"
+            + spanId2
+            + "', 'hex'))",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testSQLiWithStringArrayFilter() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createColumnExpression("Span.displaySpanName"));
+
+    Filter filter =
+        createInFilter(
+            "Span.displaySpanName", List.of("1') OR tenant_id = 'tenant2' and span_name IN ('1"));
+    builder.setFilter(filter);
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "SELECT span_name FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND span_name IN ('1'') OR tenant_id = ''tenant2'' and span_name IN (''1')",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithLikeOperator() {
+    Builder builder = QueryRequest.newBuilder();
+    Expression spanId = createColumnExpression("Span.id").build();
+    builder.addSelection(spanId);
+
+    Filter likeFilter =
+        Filter.newBuilder()
+            .setOperator(Operator.LIKE)
+            .setLhs(spanId)
+            .setRhs(createStringLiteralValueExpression("042e5523ff6b2506"))
+            .build();
+    builder.setFilter(likeFilter);
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "SELECT encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND span_id like decode('042e5523ff6b2506', 'hex')",
```

Comment:
The `encode` function is used only with bytes fields, right? So, if the field would have `service_name`, query would have been `service_name like 'frontend' ` or ``service_name like frontend`?

Reply:
Yes, encode/decode is only for bytes field